### PR TITLE
DDF-2374: Added metacard tags for valid/invalid metacards.

### PR DESCRIPTION
**Pull request by AzGoalie**
#### What does this PR do?
Adds tags (VALID or INVALID) to a metacard when it is validated.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@harrison-tarr @emanns95 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@lessarderic
#### How should this be tested?
Ingest some data that will run through a validator and verify in the search UI that it has the metacard tag VALID. Repeat with an invalid data and verify that it contains INVALID when show invalid metacards is enabled in the admin ui.
#### What are the relevant tickets?
DDF-2374
#### Checklist:
- [ ] Documentation Updated
- [x ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java [hunk 1]
```diff
@@ -115,6 +121,15 @@ public UpdateRequest process(UpdateRequest input)
                     getValidationProblems(validatorName, e, errors, warnings, counter);
                 }
             }
+
+            List<Serializable> tags = new ArrayList<>();
```
Comment (harrison-tarr) [inline catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java]: you could split this out and get rid of the boolean by moving the add valid tag to right after the `validator.validate()` and the add invalid to right after the `catch (ValidationException e)`

The declaration and `addAll` would have to go right before the loop (where the boolean is declared and assigned now)

Comment (coyotesqrl) [inline catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java]: You should be able to construct the list with `metacard.getTags()` and collapse these two lines to one.

**Comment by coyotesqrl** [comment]
Small nits aside, 👍

**Comment by emanns95** [comment]
LGTM, :+1:

File: catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java [hunk 2]
```diff
@@ -26,6 +26,8 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 import static org.mockito.Mockito.withSettings;
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.INVALID_TAG;
```
Comment (lessarderic) [inline catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java]: I usually tend _not_ to use the same constants as the class under test to ensure that the unit tests will fail if the values are (willingly or by mistake) changed in the code.

File: catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java [hunk 3]
```diff
@@ -115,32 +122,40 @@ public void setUp() {
...
+                errorExpectation,
+                warningExpectation,
+                expectedTags);
         verifyRequestPropertiesUnchanged(originalRequest, filteredRequest);
     }
 
     private void verifyMetacardErrorsAndWarnings(List<Metacard> filteredMetacards,
-            Consumer<Attribute> errorExpectation, Consumer<Attribute> warningExpectation) {
+            Consumer<Attribute> errorExpectation, Consumer<Attribute> warningExpectation,
+            Consumer<Set<String>> expectedTags) {
```
Comment (lessarderic) [inline catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java]: I'm all for lamdas, but in this case couldn't we just have used a simple `String[] expectedTags` and done an `assertThat(metacard.getTags(), hasItems(expectedTags))`?

**Comment by lessarderic** [comment]
Other than the two minor comments I left, LGTM 👍

File: catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java (hunk 2 above)
Comment (AzGoalie) [inline catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java]: Fixed

File: catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java (hunk 3 above)
Comment (AzGoalie) [inline catalog/plugin/catalog-plugin-metacard-validation/src/test/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPluginTest.java]: Changed so the methods take a string and made the two tags a constant to pass in.

**Comment by emanns95** [comment]
Passed hero :+1:

**Pull request merged**